core/gate.cc

```diff
@@ -136,8 +162,29 @@ void Gate::RemoveHook(const std::string &name) {
   }
 }
 
+GateHook *Gate::FindHookByClass(const std::string &class_name) {
+  for (const auto &hook : hooks_) {
+    if (hook->class_name() == class_name) {
+      return hook;
+    }
+  }
+  return nullptr;
+}
+
+void Gate::RemoveHookByClass(const std::string &class_name) {
+  for (auto it = hooks_.begin(); it != hooks_.end(); ++it) {
```

Comment:
`std::remove_if`? (although it changes the semantics from "remove first one" to "remove all of them")

Reply:
There seems no code that calls this method, so I thought the semantic change wouldn't matter. Anyway, it would be good to comment the expected behavior in the header file.

Reply:
I think removing all the hooks with the specified class would be more appropriate. Thanks!